connector/src/snapshot.rs

```diff
@@ -31,19 +43,29 @@ pub async fn get_snapshot_gpa(
         account_config: account_info_config.clone(),
     };
 
-    info!("requesting snapshot {}", program_id);
+    info!("requesting gpa snapshot {}", program_id);
     let account_snapshot = rpc_client
         .get_program_accounts(program_id.clone(), Some(program_accounts_config.clone()))
         .await
         .map_err_anyhow()?;
-    info!("snapshot received {}", program_id);
-    Ok(account_snapshot)
+    info!("gpa snapshot received {}", program_id);
+
+    match account_snapshot {
+        OptionalContext::Context(snapshot) => {
+            let snapshot_slot = snapshot.context.slot;
+            Ok(SnapshotProgramAccounts {
+                slot: snapshot_slot,
+                accounts: snapshot.value,
+            })
+        }
+        OptionalContext::NoContext(_) => anyhow::bail!("bad snapshot format"),
+    }
 }
 
 pub async fn get_snapshot_gma(
     rpc_http_url: String,
     ids: Vec<String>,
-) -> anyhow::Result<solana_client::rpc_response::Response<Vec<Option<UiAccount>>>> {
+) -> anyhow::Result<SnapshotMultipleAccounts> {
```

Comment:
Same here may break dependencies.

Reply:
would risk to break it. the signature is so bad